server/http/interceptors/interceptors.go

```diff
@@ -66,20 +75,23 @@ func getContentSecurityPolicyHeaderValue(nonce string) string {
 		"report-to " + contentSecurityPolicyReportingEndpointName,
 		"report-uri " + csp.ReportingEndpoint,
 		// libsodium.js requires 'wasm-unsafe-eval' to avoid a fallback to asm.js.
-		fmt.Sprintf("script-src 'nonce-%s' 'strict-dynamic' 'wasm-unsafe-eval' 'self' https: 'unsafe-inline'", nonce),
-		fmt.Sprintf("style-src 'nonce-%s' 'self' https://fonts.googleapis.com/css", nonce),
+		fmt.Sprintf("script-src %s 'strict-dynamic' 'wasm-unsafe-eval' 'self' https: 'unsafe-inline'", nonceSrc),
+		fmt.Sprintf("style-src %s 'self' https://fonts.googleapis.com/css 'unsafe-inline'", styleNonceSrc),
 	}, ";")
 }
 
-func setContentSecurityPolicy(h http.Header) string {
+func setContentSecurityPolicy(h http.Header, u *url.URL) string {
 	nonceBytes := make([]byte, 16)
 	_, err := rand.Read(nonceBytes)
 	if err != nil {
 		panic(fmt.Sprintf("Failed to generate nonce: %s", err))
 	}
 	nonce := base64.StdEncoding.EncodeToString(nonceBytes)
+	// Allow inline styles for /code/ to support the Monaco editor.
+	// https://github.com/microsoft/monaco-editor/issues/271
+	allowInlineStyles := u.Path == "/code" || strings.HasPrefix(u.Path, "/code/")
```

Comment:
This server-side path check won't work unfortunately, because we have client-side navigation. Maybe we could gate this behind whether the code feature is enabled or not? I think it's behind a flag already